src/health_misinfo_shared/youtube_api.py

```diff
@@ -107,28 +107,24 @@ def load_texts(folder) -> list[dict]:
 def form_chunks(transcript: list[dict]) -> Iterator[dict]:
     """Split/merged a list of sentences into series of overlapping text chunks.
     Each chunk is a dict containing the text and the start/end timestamps"""
-    current_chunk_text = ""
-    current_chunk_start_offset = 0.0
-    current_chunk_end_offset = 0.0
+    def chunk_length(chunk: list[dict]) -> int:
+        return sum((len(c["sentence_text"]) for c in chunk))
+
+    current_chunk = []
     for sentence in transcript:
-        current_chunk_text += sentence["sentence_text"] + " "
-        if len(current_chunk_text) > 1500:
-            current_chunk_end_offset = sentence["start"]
```

Comment:
The sentence has already been appended here, so this end offset is off by one sentence.